automake/post/rules/makedirs.am

```diff
@@ -54,16 +61,37 @@ $(foreach objects,$(filter %_OBJECTS,$(.VARIABLES)),\
    $(foreach object,$($(objects)),\
    $(eval $(call makedirs-depends,$(object)))))
 
-makedirs: force
+makedirs: $(makedirs_id)
 	$(foreach dir,$(MAKEDIRS),$(call makedirs-make,$(dir)))
 
+$(makedirs_id):
+	$(NL_V_AT)touch $@
+	$(NL_V_AT)$(RM) $(filter-out $@,$(wildcard makedirs_*))
+
+.PHONY: makedirs_clean
```

Comment:
Here, I think we can just add makedirs_id to CLEANFILES:

```
CLEANFILES += $(makedirs_id)
```

Reply:
Or, if it needs to be more complicated than that, add a `clean-local` hook and run the `$(NL_V_AT)$(RM) $(wildcard makedirs_*)` on that hook.

Reply:
> Here, I think we can just add makedirs_id to CLEANFILES:
> 
> ```
> CLEANFILES += $(makedirs_id)
> ```

ah, yes, that'd be **_much_** better

Reply:
hm, CLEANFILES is insufficient, failing 'make distcheck'

Reply:
ok, yeah.  CLEANFILES and DISTCLEANFILES are evaluated on another $(MAKE) step, so the PID changes :-(

will fix in next round, this one has to stay this way for now

Reply:
actually, I'm going ahead with the better stuff right now....  hold on merging this one

Reply:
ok, all set.

this came out better than I thought it would

still have `make -jXX` to fix

Reply:
but I think I can fix that elsewhere

Reply:
> Or, if it needs to be more complicated than that, add a `clean-local` hook and run the `$(NL_V_AT)$(RM) $(wildcard makedirs_*)` on that hook.

I missed this option, will try it

Reply:
> > Or, if it needs to be more complicated than that, add a `clean-local` hook and run the `$(NL_V_AT)$(RM) $(wildcard makedirs_*)` on that hook.
> 
> I missed this option, will try it

No worries, by parallel PR has it functional at https://github.com/project-chip/connectedhomeip/pull/465/files

Reply:
I just tried it in my tree and distcheck fails